confgenerator/otel/modular.go

```diff
@@ -164,6 +174,9 @@ func (c ModularConfig) Generate() (string, error) {
 			continue
 		}
 		for i, processor := range p {
+			if pl.ResourceOverride != nil && processor.Type == "resourcedetection" {
```

Comment:
Why is this needed? The resource detection processors shouldn't be in `p` to begin with.

Reply:
Because OTLP receiver injects a resource detection processor (named resourcedetection/otlp_0) in the config

https://github.com/GoogleCloudPlatform/ops-agent/blob/022c5dcaeba98de99b210aa2f7b81647071b6dfb/apps/otlp.go#L53

There might be other receivers doing the same.

Reply:
I was experimenting with various receivers and at some point discovered a resourcedetection receiver in the config. I didn't know where it came from and decided to simply remove  at the config generation phase. Turns out it comes from the otlp receiver when the metrics_mode is set to googlemanagedprometheus. I updated apps/otlp.go and replaced resourcedetection processor with otel.ResourceTransform() when on BMS.  PTAL ?